plugins/transformer/basic/client/recordupdater/record_updater.go

```diff
@@ -158,6 +158,54 @@ func (r *RecordUpdater) ObfuscateSensitiveColumns() (arrow.Record, error) {
 	return r.ObfuscateColumns(sensitiveColumnsArr)
 }
 
+func (r *RecordUpdater) DropRows(columnNames []string, value string) (arrow.Record, error) {
+	cols := r.record.Columns()
+
+	rowsToDrop := make([]int, 0, r.record.NumRows())
+	for j, column := range cols {
+		if !slices.Contains(columnNames, r.record.ColumnName(j)) {
+			continue
+		}
+		for i := range column.Len() {
+			if slices.Contains(rowsToDrop, i) {
+				continue
+			}
+			if (!column.IsValid(i) && value == "") || (column.IsValid(i) && column.ValueStr(i) == value) {
+				rowsToDrop = append(rowsToDrop, i)
+			}
+		}
+	}
+	if len(rowsToDrop) == 0 {
+		return r.record, nil
+	}
+
+	newRowCount := int(r.record.NumRows()) - len(rowsToDrop)
+	newCols := make([]arrow.Array, len(cols))
+	newRow := 0
+	for _, arr := range cols {
+		builder := array.NewBuilder(memory.DefaultAllocator, arr.DataType())
+		for j := 0; j < arr.Len(); j++ {
+			if slices.Contains(rowsToDrop, j) {
```

Comment:
Using a `map` for `rowsToDrop` might be more suitable

Reply:
Done in [f830fce](https://github.com/cloudquery/cloudquery/pull/20917/commits/f830fced38dfc49ee36a44af2088bf5dca43290b)